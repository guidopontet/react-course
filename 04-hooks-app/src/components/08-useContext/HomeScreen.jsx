import React, { useContext } from 'react'

import { UserContext } from './UserContext'

export const HomeScreen = () => {
  const { user, setUser } = useContext(UserContext);

  return (
    <div className='container my-3'>
      <h1>HomeScreen</h1>
      <hr />

      <pre>{ JSON.stringify(user, null, 4) }</pre>
    </div>
  )
}
