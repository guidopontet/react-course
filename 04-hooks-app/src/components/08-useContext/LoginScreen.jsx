import React, { useContext } from 'react'

import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);

  return (
    <div className='container my-3'>
      <h1>LoginScreen</h1>
      <hr />

      <button className='btn btn-primary'
              onClick={() => {
                setUser({
                  id: '123',
                  name: 'John Doe',
                });
              }}>
        Login
      </button>
    </div>
  )
}
