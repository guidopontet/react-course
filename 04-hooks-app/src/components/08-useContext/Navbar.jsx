import React from 'react'
import { NavLink } from 'react-router-dom'

export const Navbar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark px-3">
      <a className="navbar-brand" href="#">useContext</a>

      <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
        <div className="navbar-nav">
          <NavLink to="/" className="nav-item nav-link">Home</NavLink>
          <NavLink to="/about" className="nav-item nav-link">About</NavLink>
          <NavLink to="/login" className="nav-item nav-link">Login</NavLink>
        </div>
      </div>
    </nav>
  )
}
