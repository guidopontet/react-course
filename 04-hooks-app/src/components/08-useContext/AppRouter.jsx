import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';

import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { Navbar } from './Navbar';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Navbar />

        <Routes>
          <Route path="/" element={<HomeScreen />} />

          <Route path="/about" element={<AboutScreen />} />
          <Route path="/login" element={<LoginScreen />} />

          <Route path="*" element={<HomeScreen />} />
        </Routes>
      </div>
    </Router>
  );
};
