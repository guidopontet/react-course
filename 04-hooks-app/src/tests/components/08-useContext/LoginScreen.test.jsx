import { mount } from "enzyme";

import { LoginScreen } from "../../../components/08-useContext/LoginScreen";
import { UserContext } from "../../../components/08-useContext/UserContext";

describe('LoginScreen', () => {
  const setUser = jest.fn();

  const wrapper = mount(
    <UserContext.Provider value={
      { setUser }
    }>
      <LoginScreen />
    </UserContext.Provider>
  );

  test('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call setUser with expected argument', () => {
    wrapper.find('button').prop('onClick')();

    expect(setUser).toHaveBeenCalledWith({
      id: '123',
      name: 'John Doe',
    });
  });
});