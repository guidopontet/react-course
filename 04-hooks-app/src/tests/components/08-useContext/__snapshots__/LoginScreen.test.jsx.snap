// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should render 1`] = `
<div
  className="container my-3"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
